hoomd/md/pytest/test_alj_conservation.py

```diff
@@ -0,0 +1,115 @@
+import numpy as np
+import pytest
+
+import hoomd
+from hoomd import md
+
+
+@pytest.mark.validation
+def test_conservation(simulation_factory, lattice_snapshot_factory):
+    # For test, use a unit area hexagon.
+    coxeter = pytest.importorskip("coxeter",
+                                  "coxeter is required for this test.")
+
+    particle_vertices = np.array([[6.20403239e-01, 0.00000000e+00],
+                                  [3.10201620e-01, 5.37284966e-01],
+                                  [-3.10201620e-01, 5.37284966e-01],
+                                  [-6.20403239e-01, 7.59774841e-17],
+                                  [-3.10201620e-01, -5.37284966e-01],
+                                  [3.10201620e-01, -5.37284966e-01]])
+    hexagon = coxeter.shape.ConvexPolygon(particle_vertices)
+
+    circumcircle_diameter = 2 * hexagon.circumcircle_radius
+
+    # Just initialize in a simple cubic lattice.
+    sim = simulation_factory(
+        lattice_snapshot_factory(a=4 * circumcircle_diameter,
+                                 n=10,
+                                 dimensions=2))
+
+    # Initialize moments of inertia since original simulation was HPMC.
+    mass = hexagon.area
+    # https://math.stackexchange.com/questions/2004798/moment-of-inertia-for-a-n-sided-regular-polygon # noqa
+    moment_inertia = ((mass * circumcircle_diameter**2 / 6) *
+                      (1 + 2 * np.cos(np.pi / hexagon.num_vertices)**2))
+    with sim.state.cpu_local_snapshot as snapshot:
+        snapshot.particles.mass[:] = mass
+        snapshot.particles.moment_inertia[:] = np.array([0, 0, moment_inertia])
+        # Not sure if this should be incircle or circumcircle;
+        # probably doesn't matter based on current usage, but may
+        # matter in the future for the potential if it's modified
+        # to actually use diameter.
+        snapshot.particles.diameter[:] = circumcircle_diameter
+    kT = 0.3
+    sim.state.thermalize_particles(kT, 43)
+
+    # Create box resize updater
+    packing_fraction = 0.4
+    final_area = hexagon.area * sim.state.N_particles / packing_fraction
+    L_final = np.sqrt(final_area)
+    final_box = hoomd.box.square(L_final)
+
+    n_compression_start = 1e4
+    n_compression_end = 1e5
+    n_compression_total = n_compression_end - n_compression_start
+    n_total = 1e6
```

Comment:
How long does it take to run this test? If more than 2 minutes, we will need to move this test to the separate hoomd validation test suite that needs to be developed.

Reply:
We will likely need to move this. I have not run it locally to check.

Reply:
I have no idea how long this takes to run, TBH. I just ported Vyas's validation test to v3 and "called it good". Likely, this will need to be part of the separate validation suite.

Reply:
IIRC this test is not too slow and took around 2 minutes on V100s. If I'm remembering the correct test, it was definitely sub 3 minutes, but may have been slightly over 2.